spec/support/shared_active_scaffold_page_behavior.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# Very basic smoke test of ActiveScaffold components; click on each of
+# the actions and make sure it doesn't explode.
+shared_examples 'an ActiveScaffold page' do |actions, page_text|
+  describe 'the actions' do
+    before { visit(url) }
+
+    include_context 'log in super admin'
+
+    actions.each do |action|
+      describe action do
+        before do
+          first('td.actions').click_link action
+          page.driver.browser.switch_to.alert.accept if action == 'Delete'
+        end
+
+        it 'leaves us on the page' do
+          expect(page).to have_text(page_text)
+        end
+      end
+    end
+  end
+end
+
+shared_examples 'a Create New' do
+  context 'when a super admin is logged in', :js do
+    include_context 'log in super admin'
+
+    before do
+      visit(url)
+      click_link 'Create New'
+      sleep(0.5)
+    end
+
+    it 'pops up the create panel' do
+      expect(find('form.create')).not_to be_nil
+    end
+  end
+end
+
+shared_examples 'a Search' do
+  context 'when a super admin is logged in', :js do
+    include_context 'log in super admin'
+    before do
+      visit(url)
+      click_link 'Search'
+      sleep(0.5)
+    end
+
+    it 'pops up the search panel' do
+      expect(find('form.search')).not_to be_nil
+    end
+  end
```

Comment:
After looking into the specs more, I would actually simplify and consolidate these by removing the `shared_examples 'a Search'` and `shared_examples `A Create New` entirely. For pages with customized fields for each, we would be better served by custom specs. For the more general cases, where we just want to make sure nothing blows up, we can leverage the spec for the Active Scaffold actions:
```ruby
spec/support/shared_active_scaffold_page_behavior.rb
actions.each do |action|
      describe action do
        before do
          click_link action # remove the CSS scope
```
```ruby
 it_behaves_like 'an ActiveScaffold page', ['Edit', 'Delete', 'Show', 'Create New', 'Search'], 'Foos'
```

Reply:
Removed.